tests/phpunit/api/v3/OptionValueCHTest.php

```diff
@@ -192,4 +192,91 @@ public function testCreateOptionValueChange() {
     $this->assertEmpty($updatedMap['values']);
   }
 
+  public function testContributionOnOptionvalueCHValueChange() {
+    $chFund = $this->callAPISuccess('OptionValue', 'create', [
+      'label' => 'Test Created CH Fund 1',
+      'option_group_id' => 'ch_fund',
+      'value' => 'CH+99999',
+    ]);
+    $chFundMap = $this->callAPISuccess('OptionValueCH', 'get', ['value' => 'CH+99999']);
+    $this->assertEquals($this->unallocatedFund['id'], $chFundMap['financial_type_id']);
+    // change the OptionValueCH mapping, by updating the financial_type_id
+    $this->callAPISuccess('OptionValueCH', 'create', ['financial_type_id' => $this->fund['id'], 'id' => $chFundMap['id']]);
+
+
+    $contributionID = $this->callAPISuccess('CHContribution', 'create', [
+      'receive_date' => date('Ymd'),
+      'total_amount' => 100.00,
+      'payment_instrument_id' => 1,
+      'source' => 'SSF',
+      'contribution_status_id' => 1,
+      'ch_fund' => 'CH+99999'
+    ])['id'];
+    $contribution = $this->callAPISuccess('Contribution', 'getsingle', [
+      'id' => $contributionID,
+    ]);
+    // ensure that contribution Fund is assigned to 'Test Created Fund'
+    $this->assertEquals($this->fund['id'], $contribution['financial_type_id']);
+
+    // change CH Fund option value
+    $this->callAPISuccess('OptionValueCH', 'create', ['value' => 'CH+1000000', 'id' => $chFundMap['id']]);
+
+    // ensure that option value is changed
+    $actualOptionValue = $this->callAPISuccess('OptionValue', 'getvalue', ['id' => $chFund['id'], 'return' => 'value']);
+    $this->assertEquals('CH+1000000', $actualOptionValue);
+
+    // ensure that contribution's CH Fund is also updated
+    $contribution = $this->callAPISuccess('Contribution', 'getsingle', [
+      'id' => $contributionID,
+      'return' => ['custom_' . $this->customField['id'], 'financial_type_id'],
+    ]);
+    // ensure that the value of the custom field is changed
+    $this->assertEquals('CH+1000000', $contribution['custom_' . $this->customField['id']]);
+    $this->assertEquals($this->fund['id'], $contribution['financial_type_id']);
+
+    // delete created values
+    $this->callAPISuccess('contribution', 'delete', ['id' => $contributionID]);
+    $this->callAPISuccess('OptionValue', 'delete', ['id' => $chFund['id']]);
+    $updatedMap = $this->callAPISuccess('OptionValueCH', 'get', []);
+    $this->assertEmpty($updatedMap['values']);
+  }
+
+  public function testContributionOnOptionvalueCHFundChange() {
+      $chFund = $this->callAPISuccess('OptionValue', 'create', [
+      'label' => 'Test Created CH Fund 1',
+      'option_group_id' => 'ch_fund',
+      'value' => 'CH+99999',
+    ]);
+
+    $contributionID = $this->callAPISuccess('CHContribution', 'create', [
+      'receive_date' => date('Ymd'),
```

Comment:
same here missing contact_id parameter

Reply:
@seamuslee001 added now. Thanks for pointing this out

sigh, how could I missed such essential attribute :(